s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -0,0 +1,108 @@
+package ch.cyberduck.core.sts;/*
+ * Copyright (c) 2002-2023 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.Credentials;
+import ch.cyberduck.core.Host;
+import ch.cyberduck.core.PasswordCallback;
+import ch.cyberduck.core.aws.CustomClientConfiguration;
+import ch.cyberduck.core.exception.LoginCanceledException;
+import ch.cyberduck.core.exception.LoginFailureException;
+import ch.cyberduck.core.ssl.ThreadLocalHostnameDelegatingTrustManager;
+import ch.cyberduck.core.ssl.X509KeyManager;
+import ch.cyberduck.core.ssl.X509TrustManager;
+
+import ch.cyberduck.core.LoginOptions;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSSessionCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.AnonymousAWSCredentials;
+import com.amazonaws.auth.profile.internal.AllProfiles;
+import com.amazonaws.auth.profile.internal.BasicProfile;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityResult;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AssumeRoleWithWebIdentitySTSCredentialsConfigurator extends AWSProfileSTSCredentialsConfigurator {
```

Comment:
Extract common base class for existing `AWSProfileSTSCredentialsConfigurator` and new implemenation.

Reply:
You mean to create a new abstract class and both classes extend this abstract class? I was just tryting to do this. In my opinion the abstract class would only contain the logger, the trust- and key-manager. The code is very different in these two classes. Is that what you want?

Reply:
It should be possible to share the token service from `getTokenService`.

Reply:
I will create a new base class containing the following fields. Is that what you mean? The getTokenService-Method differs quite a lot between both classes so their implementation will be in the subclasses.

public abstract class AbstractSTSCredentialsConfigurator implements CredentialsConfigurator {
    private static final Logger log = LogManager.getLogger(AbstractSTSCredentialsConfigurator.class);

    protected final X509TrustManager trust;
    protected final X509KeyManager key;
    protected AWSSecurityTokenService service;

Reply:
Is now implemented like this